lib/consoletest.pm

```diff
@@ -48,13 +49,19 @@ Method executed when run() finishes and the module has result => 'fail'
 
 =cut
 sub post_fail_hook {
-    my ($self) = shift;
+    my ($self) = @_;
     $self->SUPER::post_fail_hook;
     select_console('log-console');
     $self->remount_tmp_if_ro;
     $self->export_logs_basic;
     # Export extra log after failure for further check gdm issue 1127317, also poo#45236 used for tracking action on Openqa
     $self->export_logs_desktop;
+
+    # Destroy the public cloud instance in case of fatal test failure
+    if (is_publiccloud() && $self->{result} eq 'fail') {
```

Comment:
as we are in `post_fail_hook` which by design triggered only when test is failing I think you can cut second part - 
```suggestion
    if (is_publiccloud()) {
```

Reply:
I might be wrong but I still don't think so. The tests may fail without being fatal.
The `fatal` flag means that the test module will fail the whole test run.

Reply:
> I might be wrong but I still don't think so. The tests may fail without being fatal.
> The `fatal` flag means that the test module will fail the whole test run.

yeah you absolutely right I just come to write something similar . but still your current code not follows what you saying here. You need to do something like this : 
```
my $flags = $self->test_flags();
 if (is_publiccloud() && $flags->{fatal}) {
```
 to follow what you stating here

Reply:
Oh, I'm sorry. Thank you Anton, it should be solved now.